shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -35,35 +63,94 @@
     private static final float SCROLL_POSITION_CAP_FOR_INFINITY = 70000.0f;
     private static final int ROOT_NODE_ID = 0;
 
-    private final FlutterView owner;
+    /// Value is derived from ACTION_TYPE_MASK in AccessibilityNodeInfo.java
+    private static int FIRST_RESOURCE_ID = 267386881;
+
+    // Real Android View, which internally holds a Flutter UI.
+    @NonNull
+    private final View rootAccessibilityView;
+
+    // Direct interface between Flutter's Android embedding and the Flutter framework.
+    @NonNull
+    private final FlutterJNI flutterJNI;
+
+    // The accessibility communication API between Flutter's Android embedding and
+    // the Flutter framework.
+    @NonNull
     private final AccessibilityChannel accessibilityChannel;
+
+    // The top-level Android View within the containing Window.
+    // TODO(mattcarroll): why do we need this? and why are we caching it?
+    @NonNull
     private final View decorView;
-    private Map<Integer, SemanticsObject> objects;
-    private Map<Integer, CustomAccessibilityAction> customAccessibilityActions;
+
+    // TODO(mattcarroll): what exactly is a semantic object, and why do we cache them?
+    @NonNull
+    private final Map<Integer, SemanticsObject> semanticsObjects;
+
+    // TODO(mattcarroll): what makes a CustomAccessibilityAction different than a regular
+    //                    AccessibilityAction. why do we cache them?
+    @NonNull
+    private final Map<Integer, CustomAccessibilityAction> customAccessibilityActions;
```

Comment:
Android AccessibilityNodeInfo comes with a limited set of actions that may be performed, for example scrollUp, scrollDown, longPress.

Certain UI styles don't lend themselves well to interaction through these values, for example a re-orderable list. A custom accessibility action allows you to provide a name + callback like "move item to end" and have Android expose this through the local context menu.

Similarly to the caching the full nodes, these may not be updated but android can ask for them at any time.